.github/workflows/node.js.yml

```diff
@@ -39,7 +39,9 @@ jobs:
         run: yarn lint
 
       - name: 🏗️ Build app
-        run: yarn build --configuration production
+        run: yarn build --configuration $buildEnvironment
+        env: 
+          buildEnvironment: ${{ github.ref == 'refs/heads/staging' && 'staging' || 'production' }}
```

Comment:
for the checks & build test I would always have it build production, that's the important system to make sure things build correctly for